.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,203 @@
+            pmd-cataloger/build/test-results
+      - name: Upload sfge test artifacts
+        if: ${{ always() }}
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ runner.os }}-sfge-results
+          path: |
+            sfge/build/reports
+            sfge/build/test-results
+      - name: Upload cli-messaging test artifacts
+        if: ${{ always() }}
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ runner.os }}-cli-messaging-results
+          path: |
+            cli-messaging/build/reports
+            cli-messaging/build/test-results
+  # Step 2B: Run the smoke tests
+  smoke-tests:
+    strategy:
+      # By default, if any job in a matrix fails, all other jobs are immediately canceled. This makes the jobs run
+      # to completion instead.
+      fail-fast: false
+      matrix:
+        node: ${{ fromJson(inputs.node-matrix) }}
+        os: [{vm: ubuntu-latest, exe: .sh}, {vm: windows-2019, exe: .cmd}]
+    runs-on: ${{ matrix.os.vm }}
+    needs: build-dependencies
+    steps:
+      # Check out the code.
+      - uses: actions/checkout@v2
+      # Make sure we're using the right versions of Node and Java.
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node }} # Node version is a matrix.
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '11' # For now, Java version is hardcoded. It could be changed to a matrix in the future.
+      # Install Salesforce CLI via NPM
+      - run: npm install -g sfdx-cli
+      # Download and install the tarball artifact built during setup.
+      - uses: actions/download-artifact@v3
+        id: download
+        with:
+          name: smoke-test-tarball
+          # Download the tarball to subdirectory of HOME, so it's guaranteed
+          # to be in a place where the installation command can see it.
+          path: ~/downloads/tarball
+      - run: echo ${{ steps.download.outputs.download-path }}
+      - name: Install Tarball
+        shell: bash
+        run: |
+          # We need to determine the tarball's name first.
+          TARBALL_NAME=$(ls ~/downloads/tarball | grep salesforce-sfdx-scanner-[0-9]*\\.[0-9]*\\.[0-9]*\\.tgz)
+          # We need to determine where the tarball is located.
+          # Get the path to the folder, and swap out any backslashes for forward slashes (needed for Windows).
+          RAW_TARBALL_PATH=`echo '${{ steps.download.outputs.download-path }}' | tr '\\' '/'`
+          # If the path starts with C:, we need to rip that off (needed for Windows).
+          ADJUSTED_TARBALL_PATH=`[[ $RAW_TARBALL_PATH = C* ]] && echo $RAW_TARBALL_PATH | cut -d':' -f 2 || echo $RAW_TARBALL_PATH`
+          # Pipe in a `y` to simulate agreeing to install an unsigned package. Use a URI of the file's full path.
+          echo y | sfdx plugins:install "file://${ADJUSTED_TARBALL_PATH}/${TARBALL_NAME}"
```

Comment:
Can windows fail with the `/` in the path instead of `\`?

Reply:
In general, for this block, how about gating with an "if os == windows" check?

Reply:
I never saw it fail, so I think it should be fine without.